### DISCUSSION TITLE
Cannot find module 'util' in blind_threshold_bls.js

### DISCUSSION THREAD

**Discussion by nestorbonilla** [celo-org/social-connect#252] [Help (Q&A)]
I’m building a dApp with SocialConnect on a Nextjs app without contractKit following the ethers example [code](https://github.com/celo-org/SocialConnect/blob/main/examples/ethers.ts). When I execute `OdisUtils.Identifier.getObfuscatedIdentifier` from the `@celo/identity package`, I’m getting a `“cannot find module ‘util’` error, and I verified I’m passing all the expected arguments.
Is there any package I need to add or any additional configuration to solve this issue? Here’s the respective [repo](https://github.com/nestorbonilla/celo-social-connect-dapp).

<img width="1350" alt="error_1" src="https://github.com/celo-org/SocialConnect/assets/3916389/402df9c5-e9fe-4069-bfc5-d3160ddca5b7">

The error comes from blind_threshold_bls.js on line number 4. We also tried to install the util lib (just in case), but it didn’t work.

![error_2](https://github.com/celo-org/SocialConnect/assets/3916389/4dc3e9fb-364c-434a-93a0-226ac62841ae)

**Comment by alecps** [comment]
Unfortunately I don't have any special insight here, other than that it looks like the util module is not automatically available in Nextjs the way we would expect it to be. A quick google of `cannot find module ‘util'` shows that this problem comes up for various reasons and some stackoverflow threads suggest fixes involving adding fields to tsconfig and similar. If you feel you've exhausted everything, it might be worth querying the broader eng channel to see if anyone has seen this error before (perhaps someone with webdev / frontend experience like Aaron?). If more in depth support is needed to unblock this lmk and we can work to figure out how to resource some dev days for that

**Comment by therealharpaljadeja** [comment]
`util` is part of `node` and since the code you are using for blinding is being used in the client side of Next.js, it can't find it.

Instead follow the `WebBlsBlindingClient` steps [here](https://github.com/celo-org/SocialConnect/blob/main/privacy.md#web)

**Reply by Temitope3665** [reply]
I followed the same process mentioned here, but i keep getting that same error
